docs/4.tools/fastauth-sdk.md

```diff
@@ -18,68 +18,245 @@ The FastAuth system is comprised of 3 main components:
+```bash
+cargo install near-cli-rs
+NEAR_ENV=mainnet
+near account create-account fund-later use-auto-generation save-to-folder ~/.near-credentials/implicit
+```
+
+This should output something like:
+
+```bash
+The file "~/.near-credentials/implicit/275f14eecb0afcb1f46f2b71b7933afd2de6d4ae8b08e9b11fc538a5a81406b7.json" was saved successfully
+```
 
-You'll also need to add your Firebase Project ID as the environment variable `firebaseProjectID`. You can do so by following [these steps](https://cloud.google.com/run/docs/configuring/services/environment-variables).
+In this example. `275f14eecb0afcb1f46f2b71b7933afd2de6d4ae8b08e9b11fc538a5a81406b7` is your funded account. We'll refer to this as `$FUNDED_ACCOUNT` from now on.
 
-### Integrating your frontend
+Send some NEAR to this address.
 
-As a last step, you'll need to install and instantiate the FastAuth npm module:
+#### Adding multiple keys (Recommended)
+
+This account has been created with one key. However, due to [this](https://near.zulipchat.com/#narrow/stream/295302-general/topic/.E2.9C.94.20The.20trouble.20with.20nonces/near/389649443), you should create an account with `N` keys where `N` is the number of requests you expect to get in a second, at peak load.
+
+To generate an additional key, run the following command:
 
 ```bash
-npm install --save @pagoda/fastauth
+near account add-key $FUNDED_ACCOUNT grant-full-access autogenerate-new-keypair save-to-keychain network-config mainnet sign-with-access-key-file ~/.near-credentials/implicit/$FUNDED_ACCOUNT.json send
 ```
 
-And then, on your app:
 
-```js
-import { FastAuth } from "@pagoda/fastauth";
+#### Deploying the relayer
+
+Run the following command:
+
+```bash
+git clone https://github.com/near/pagoda-relayer-rs
+```
+
+Go to `config.toml` and change:
+
+```toml
+network = "mainnet"
+num_keys = 3  # correlates to the number of keys in `keys_filenames`. Will be optional in the future.
+relayer_account_id = "$FUNDED_ACCOUNT"
+keys_filenames = [
+    # The original account
+    "~/.near-credentials/mainnet/$FUNDED_ACCOUNT.json",
 
-FastAuth.setupFastAuth(signer_url) // replace 'signer_url' with the URL of your signer app instance
+    # Other keys you've optionally created. This will allow rotating through each key as to avoid nonce races.
+    "~/.near-credentials/mainnet/$FUNDED_ACCOUNT/ed25519_4ryLkp4AuzBD8yuyRJKb91hvHZ4zgqouWcJzu1gNEvLv.json",
+    "~/.near-credentials/mainnet/$FUNDED_ACCOUNT/ed25519_7K3jF8Ft5dKFEPYRH1T4mncvsZGgSoGKsvsnnKEmqubT.json"
+]
 ```
+
+Optionally, if we need to generate additional access keys for the `$FUNDED_ACCOUNT`, run the following command N times. Note that this will generate keys for implicit accounts, but we'll then tie them to `$FUNDED_ACCOUNT`.
```

Comment:
```suggestion
Optionally, if you need to generate additional access keys for the `$FUNDED_ACCOUNT`, run the following command N times. Note that this will create keys for implicit accounts, but we'll then tie them to `$FUNDED_ACCOUNT`.
```